### PR TITLE
Dashboard selector dropdown.

### LINKED ISSUE #7637: Select none/all in dashboard
I think most UIs where you can select multiple things (files, emails, etc.) provide some way to quickly select everything, or possibly everything visible, and even more importantly, to empty the selection without having to click again on each selected item. We should consider this for the file list in the dashboard.

We could also have ways to select e.g. all notebooks, all running notebooks, etc., but I think that's less important. GMail has UI for selecting groups that we could <s>shamelessly copy</s> be inspired by:

gmail_selecting

### PR DESCRIPTION
**Pull request by jdfreder**
closes #7637

~~Before anyone asks, I tried hard to find a quick/easy way to make the checkbox a tristate (including manually overlaying fa-minus, etc...).  There are no good ways to do this.  We will either need our own control or another third party component- which I don't think we should do at this point (maybe in our next major release).~~

@ellisonbg for design review.

### CODE REVIEW

**Comment by jdfreder** [comment]
screen shot 2015-02-02 at 4 38 48 pm

**Comment by jdfreder** [comment]
screen shot 2015-02-02 at 4 39 00 pm

**Comment by jdfreder** [comment]
screen shot 2015-02-02 at 4 44 09 pm

**Comment by takluyver** [comment]
:+1: 

I would lose 'select inverse' (confusing and almost never useful) and put a divider before 'unselect all'.

File: IPython/html/templates/tree.html [hunk 1]
```diff
@@ -76,6 +76,24 @@
...
+    <span class="caret"></span>
+  </button>
+  <ul class="dropdown-menu" role="menu" aria-labelledby="tree-selector-btn">
+    <li role="presentation" class="select-all"><a role="menuitem" tabindex="-1" href="#">Select all</a></li>
+    <li role="presentation" class="select-notebooks"><a role="menuitem" tabindex="-1" href="#">Select notebooks</a></li>
+    <li role="presentation" class="select-running-notebooks"><a role="menuitem" tabindex="-1" href="#">Select running notebooks</a></li>
+    <li role="presentation" class="select-files"><a role="menuitem" tabindex="-1" href="#">Select files</a></li>
+    <li role="presentation" class="select-directories"><a role="menuitem" tabindex="-1" href="#">Select directories</a></li>
+    <li role="presentation" class="select-inverse"><a role="menuitem" tabindex="-1" href="#">Select inverse</a></li>
+    <li role="presentation" class="unselect-all"><a role="menuitem" tabindex="-1" href="#">Unselect all</a></li>
```
Comment (minrk) [inline IPython/html/templates/tree.html]: I think 'deselect' is the more usual word to use here

**Comment by ellisonbg** [comment]
Oohhh, so cool. Life @fperez I am completely lost in grant land. don't wait for me on this. I will be at the meeting tomorrow and will slowly start to plug back in a bit.

File: IPython/html/templates/tree.html (hunk 1 above)
Comment (jdfreder) [inline IPython/html/templates/tree.html]: I should nest this block too.  I was working in a non-nested state so I could easily keep track of what was new and what wasn't.

**Comment by Carreau** [comment]
> Before anyone asks, I tried hard to find a quick/easy way to make the checkbox a tristate (including manually overlaying fa-minus, etc...). There are no good ways to do this. We will either need our own control or another third party component- which I don't think we should do at this point (maybe in our next major release).

http://jsfiddle.net/5tpXc/ ?

**Comment by jdfreder** [comment]
WOW!  Thanks @Carreau !  I was way overcomplicating that.  Tristate checked in.

**Comment by jdfreder** [comment]
screen shot 2015-02-03 at 9 52 19 am

**Comment by jdfreder** [comment]
screen shot 2015-02-03 at 9 53 10 am

**Comment by jdfreder** [comment]
screen shot 2015-02-03 at 9 54 34 am

**Comment by jdfreder** [comment]
From the meeting, I'm going to iterate in person with @ellisonbg today, and then it should be ready to merge (as long as others don't object).

**Comment by jdfreder** [comment]
@ellisonbg I tried doing what we talked about with the alignment, but it just doesn't look right.

Current state:
screen shot 2015-02-03 at 3 31 48 pm

Highlighting the baseline alignment:
hline

Adding equal height blocks to show how content as a whole is aligned:
centering

**Comment by ellisonbg** [comment]
Hmmm, looks like one of the images didn't upload. Can you clarify which you
think is best?

On Tue, Feb 3, 2015 at 3:42 PM, Jonathan Frederic <notifications@github.com>
wrote:

> @ellisonbg <https://github.com/ellisonbg> I tried doing what we talked
> about with the alignment, but it just doesn't look right.
>
> Current state:
> [image: screen shot 2015-02-03 at 3 31 48 pm]
> <https://cloud.githubusercontent.com/assets/3292874/6032197/2dacc0f4-abbb-11e4-95a2-173611af409a.png>
>
> Highlighting the baseline alignment:
> [image: hline]
> <https://cloud.githubusercontent.com/assets/3292874/6032199/398190e4-abbb-11e4-8448-8bc58c6c25cd.png>
>
> Adding equal height blocks to show how content as a whole is aligned:
> [image: centering]
> <https://cloud.githubusercontent.com/assets/3292874/6032207/46773b28-abbb-11e4-8160-2854e89afff9.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/pull/7667#issuecomment-72761037>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by takluyver** [comment]
what about vertical-align: middle?

**Comment by jdfreder** [comment]
@takluyver thanks, I think that's basically what the final is (see last commit):

screen shot 2015-02-03 at 4 12 20 pm

@ellisonbg see attached

**Comment by ellisonbg** [comment]
I think this looks good. Thanks for chasing the pixels :)

On Tue, Feb 3, 2015 at 4:14 PM, Jonathan Frederic <notifications@github.com>
wrote:

> @takluyver <https://github.com/takluyver> thanks, I think that's
> basically what the final is (see last commit):
>
> [image: screen shot 2015-02-03 at 4 12 20 pm]
> <https://cloud.githubusercontent.com/assets/3292874/6032580/aded89f2-abbf-11e4-9bdc-65e96f220617.png>
>
> @ellisonbg <https://github.com/ellisonbg> see attached
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/pull/7667#issuecomment-72765136>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jdfreder** [comment]
I think this is ready to go.

**Comment by ellisonbg** [comment]
Merge and play?

**Comment by minrk** [comment]
:+1:, :shipit: @ellisonbg

**Comment by ellisonbg** [comment]
Yeah, I got to contribute by pushing the green button :) After a few weeks of grant writing, this feels good...

**Pull request merged**